### PR TITLE
feat: Create custom attributes for a contact from CRM

### PR DESCRIPTION
**Pull request by nithindavid**
**Note** - The Button refactor has to be merged before this.

image

### CODE REVIEW

File: app/javascript/dashboard/modules/contact/components/AddCustomAttribute.vue [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    />
+    <form class="row" @submit.prevent="addCustomAttribute">
+      <woot-input
+        v-model.trim="attributeName"
+        :class="{ error: $v.attributeName.$error }"
+        class="medium-12 columns"
+        :label="$t('CUSTOM_ATTRIBUTES.FORM.NAME.LABEL')"
+        :placeholder="$t('CUSTOM_ATTRIBUTES.FORM.NAME.PLACEHOLDER')"
+        @input="$v.attributeName.$touch"
+      />
```
Comment (muhsin-k) [inline app/javascript/dashboard/modules/contact/components/AddCustomAttribute.vue:15]: Show validation errors under the input box

**Comment by muhsin-k** [comment]
@nithindavid Please see the inline comments. Also, add spacing between buttons 
Screenshot 2021-05-19 at 1 30 34 PM

File: app/javascript/dashboard/modules/contact/components/AddCustomAttribute.vue [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    <woot-modal-header
+      :header-title="$t('CUSTOM_ATTRIBUTES.ADD.TITLE')"
+      :header-content="$t('CUSTOM_ATTRIBUTES.ADD.DESC')"
+    />
+    <form class="row" @submit.prevent="addCustomAttribute">
+      <woot-input
+        v-model.trim="attributeName"
+        :class="{ error: $v.attributeName.$error }"
+        class="medium-12 columns"
+        :error="$v.attributeName.$error"
```
Comment (muhsin-k) [inline app/javascript/dashboard/modules/contact/components/AddCustomAttribute.vue:12]: The `error`  prop should be the error message.

Screenshot 2021-05-21 at 10 19 02 AM

**Review by muhsin-k — APPROVED** [review]
@nithindavid Good work. Please fix the `prop` related issue

File: app/javascript/dashboard/routes/dashboard/conversation/contact/NewConversation.vue [hunk 3]
```diff
@@ -34,12 +34,17 @@ export default {
   },
   watch: {
     'contact.id'(id) {
-      this.$store.dispatch('contacts/fetchContactableInbox', id);
+      if (id) {
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/conversation/contact/NewConversation.vue:37]: Doesn't this mean that we are mounting the component even if contact is not available?

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**